In [12]:
import pandas as pd
import scipy.stats as ss

In [13]:
table = pd.read_csv("E:/XacSuatThongKe/DuLieuThucHanh/Du_lieu/hostel.csv")

a. Khách du lịch thường nghĩ rằng Tokyo là thủ đô nên giá nhà trọ sẽ cao hơn các

thành phố khác. Nhận định này có đúng cho hai thành phố Tokyo và Kyoto

tại mức ý nghĩa α = 5% không.


mu1: Giá nhà trọ trung bình tại Tokyo

mu2: Giá nhà trọ trung bình tại Kyoto

Giả thuyết cần kiểm định là: mu1 - mu2 > 0

H0: mu1 <= mu2

H1: mu1 > mu2

In [14]:
price_tokyo = table.query('City == "Tokyo"')['price_from']
price_kyoto = table.query('City == "Kyoto"')['price_from']

In [15]:
ss.ttest_ind(price_tokyo, price_kyoto, equal_var=False, alternative="greater")

TtestResult(statistic=np.float64(1.0493840858717811), pvalue=np.float64(0.14810063553206787), df=np.float64(115.02265442731779))

p-value = 0.85 > alpha nên không bác bỏ H0

Tại mức ý nghĩa 5%, không có đủ bằng chứng chứng minh giá nhà trọ trung bình tại Tokyo cao hơn Kyoto

Hay là ta có cơ sở thống kê để cho rằng giá nhà trọ ở Tokyo không cao hơn Kyoto

b. Kiểm định tại mức ý nghĩa 1% không có sự khác biệt trong trung bình điểm

đánh giá về tiện ích (facilities) và trung bình điểm đánh giá về độ sạch sẽ

(cleanliness)

mu1: Trung bình điểm đánh giá tiện ích

mu2: Trung bình điểm đánh giá độ sạch sẽ

Giả thuyết cần kiểm định: mu1 - mu2 = 0

H0: mu1 = mu2

H1: mu1 != mu2

In [16]:
ss.ttest_rel(table['facilities'], table['cleanliness'], alternative="two-sided")

TtestResult(statistic=np.float64(-9.379956003538563), pvalue=np.float64(1.7664840199466876e-18), df=np.int64(297))

p-value = 1.76 * 10^-18 < alpha nên bác bỏ H0

Tại mức ý nghĩa 1%, có đủ bằng chứng để chứng minh có sự khác biệt giữa trung bình điểm đanh giá tiện ích và độ sạch sẽ

c. Khách hàng thường sẽ đánh giá cao các nhà trọ ở thủ đô Tokyo hơn so với các

thành phố khác. Dữ liệu về điểm đánh giá (summary_score) giữa thành phố

Tokyo và Osaka có ủng hộ nhận định này không? Kết luận đưa ra tại mức ý

nghĩa α = 5%.

mu1: Điểm đánh giá trung bình của thành phố Tokyo

mu2: Điểm đánh giá trung bình của thành phố Osaka

Giả thuyết cần kiểm định: mu1 - mu2 > 0

H0: mu1 <= mu2

H1: mu1 > mu2

In [17]:
score_tokyo = table.query('City == "Tokyo"')['summary_score']
score_osaka = table.query('City == "Osaka"')['summary_score']

In [18]:
ss.ttest_ind(score_tokyo, score_osaka, equal_var=False, alternative="greater")

TtestResult(statistic=np.float64(0.9179886057315083), pvalue=np.float64(0.18008800111320855), df=np.float64(142.49663564093694))

p-value = 0.18 > alpha nên không bác bỏ H0

Tại mức ý nghĩa 5%, không có đủ bằng chứng chứng minh điểm trung bình đánh giá ở Tokyo lớn hơn Osaka

d. Osaka là một thành phố hiện đại và sôi động, còn Kyoto là một thành phố lịch

sử và văn hóa. Khách du lịch cho rằng các nhà trọ ở Kyoto sẽ an toàn hơn và

cũng có sự đồng đều về độ an toàn (security) hơn so với Osaka. Kiểm định

lại những nhận định này tại mức ý nghĩa α = 5%.

mu1: Điểm đánh giá trung bình về độ an toàn của Kyoto

mu2: Điểm đánh gái trung bình về độ an toàn của Osaka

H0: mu1 >= mu2

H1: mu1 < mu2

In [19]:
security_score_kyoto = table.query('City == "Kyoto"')['security']
security_score_osaka = table.query('City == "Osaka"')['security']

In [34]:
ss.ttest_ind(security_score_kyoto, security_score_osaka, equal_var=False, alternative="less")

TtestResult(statistic=np.float64(1.5276998562148962), pvalue=np.float64(0.9355884643537153), df=np.float64(141.26494898833218))

p-value = 0.93 > alpha nên không bác bỏ H0

Tại mức ý nghĩa 5%, có đủ bằng chứng để chứng minh điểm trung bình độ an toàn của Kyoto tốt hơn và đồng đều hơn Osaka

e. Tỷ lệ điểm đánh giá mức độ sạch sẽ (cleanliness) tại Tokyo trên 9 không thấp

hơn tỷ lệ này ở Osaka. Kiểm định lại nhận định này tại mức ý nghĩa α = 5%

p1: Tỷ lệ đánh giá mức độ sạch sẽ ở Tokyo

p2: Tỷ lệ đánh giá mức độ sạch sẽ ở Osaka

H0: p1 >= p2

H1: p1 < p2

In [21]:
from statsmodels.stats.proportion import proportions_ztest as pz
import numpy as np

In [32]:
tokyo = table.query('City == "Tokyo"')
x1 = np.sum(tokyo['cleanliness'] > 9) # 80
n1 = tokyo['cleanliness'].shape[0] # 116
print(x1)
print(n1)
# p1 = 0.69

osaka = table.query('City == "Osaka"')
x2 = np.sum(osaka['cleanliness'] > 9) # 49
n2 = osaka['cleanliness'].shape[0] # 89
print(x2)
print(n2)
# p2 = 0.55

count = np.array([x1,x2])
nobs = np.array([n1,n2])

pz(count, nobs, value=0, alternative="smaller")

80
116
49
89


(np.float64(2.043646243021458), np.float64(0.979505746289047))

p-value = 0.98 > alpha nên không bác bỏ H0
Tại mức ý nghĩa 5%, có đủ bằng chứng chứng minh tỷ lệ mức độ sạch sẽ trên 9 của thành phố Tokyo không thấp hơn osaka